src/Theme/HookPreprocess.php

```diff
@@ -53,6 +54,21 @@ public static function block(array &$variables): void
         $block = \Drupal::entityTypeManager()
             ->getStorage('block')
             ->load($variables['elements']['#id']);
+        assert($block instanceof BlockInterface);
+        $block_counter = &drupal_static('template_preprocess_block', []);
+        assert(is_array($block_counter));
         $variables['block'] = new WrappedConfigEntity($block);
+        if (!isset($block_counter[$block->getRegion()])) {
+            $block_counter[$block->getRegion()] = 1;
+        }
+        $variables['block_zebra'] = $block_counter[$block->getRegion()] % 2 ? 'odd' : 'even';
+        $variables['block_id'] = $block_counter[$block->getRegion()]++;
+        $variables['classes_array'][] = drupal_html_class('block-' . $variables['configuration']['provider']);
+        $variables['classes'] = implode(' ', $variables['classes_array']);
+        $variables['theme_hook_suggestions'][] = 'block__' . $block->getRegion();
+        $variables['theme_hook_suggestions'][] = 'block__' . $variables['configuration']['provider'];
+        $variables['theme_hook_suggestions'][] = 'block__' . $variables['configuration']['provider']
```

Comment:
This works this late in applying theme hook suggestions? Or do we need to implement a way to provide theme hook suggestions?